spec/controllers/events_controller_spec.rb

```diff
@@ -0,0 +1,87 @@
+require 'rails_helper'
+
+RSpec.describe Api::EventsController, type: :controller do
+  let(:event_attributes) { %w[id name date description status user_id] }
+
+  describe 'GET #index' do
+    it 'should get index' do
+      get :index
+      expect(response).to have_http_status(:success)
+    end
+  end
+
+  describe 'GET #show' do
+    context 'when valid' do
+      let(:event) { create(:event) }
+      it 'returns json response with event' do
+        get :show, params: { id: event.id }
+        json_response = JSON.parse(response.body)
+
+        expect(json_response.keys).to eq(event_attributes)
+        expect(json_response['name']).to eq(event.name)
+      end
+    end
+
+    context 'when invalid' do
+      it 'return unprocessable_entity status' do
+        get :show, params: { id: -1 }
+        json_response = JSON.parse(response.body)
+
+        expect(json_response['status']).to eq('unprocessable_entity')
+      end
+    end
+  end
+
+  describe 'POST #create' do
+    context 'when valid' do
+      let(:valid_params) do
```

Comment:
why not to define factory and use 
let(:event) { build(:event) } 
post :create, params: { event: event.attributes }